core/src/main/java/org/elasticsearch/search/suggest/SuggestBuilder.java

```diff
@@ -112,13 +112,18 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
-    public static SuggestBuilder fromXContent(QueryParseContext parseContext, Suggesters suggesters) throws IOException {
+    public static SuggestBuilder fromXContent(QueryParseContext parseContext, Suggesters suggesters, final boolean fromStart)
+        throws IOException {
+
         XContentParser parser = parseContext.parser();
         ParseFieldMatcher parseFieldMatcher = parseContext.parseFieldMatcher();
         SuggestBuilder suggestBuilder = new SuggestBuilder();
         String fieldName = null;
 
         XContentParser.Token token;
+        if (fromStart && parser.nextToken() != XContentParser.Token.START_OBJECT) {
```

Comment:
I see the problem but I find adding this extra "state" parameter confusing. I would either make sure that the parser is already on the start token on the caller side (I think in SearchSourceBuilder this is always the case, in tests and in RestSuggestAction we should make sure it is), or check if `parser.currentToken == null` inside the SuggestBuilder parse method and advance the token pointer if so. This btw. is what ObjectParser#parse() does by default. I think this would be the best solution.